projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -4489,4 +4861,148 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
+        this.transactions.startPending();
+        this.configureRowEditingOverlay(row);
+        this.rowEditingOverlay.open(this.rowEditSettings);
+        this.rowEditPositioningStrategy.isTopInitialPosition = this.rowEditPositioningStrategy.isTop;
+        this.rowEditingOverlay.element.addEventListener('wheel', this.rowEditingWheelHandler.bind(this));
+    }
+
+    /**
+     * @hidden
+     */
+    public closeRowEditingOverlay(commit?: boolean) {
+        this.transactions.endPending(commit);
+        this.rowEditingOverlay.element.removeEventListener('wheel', this.rowEditingWheelHandler);
+        this.rowEditPositioningStrategy.isTopInitialPosition = null;
+        this.rowEditingOverlay.close();
+    }
+
+    private showRowEditingOverlay() {
+        this.rowEditingOverlay.element.style.display = 'block';
+    }
+
+    private hideRowEditingOverlay() {
+        this.rowEditingOverlay.element.style.display = 'none';
+    }
+
+    private repositionRowEditingOverlay(row: IgxGridRowComponent) {
+        this.configureRowEditingOverlay(row);
+        if (!this.rowEditingOverlay.collapsed) {
+            this.rowEditingOverlay.reposition();
+        }
+    }
+
+    private configureRowEditingOverlay(row: IgxGridRowComponent) {
+        this.rowEditSettings.outlet = this.rowEditingOutletDirective;
+        this.rowEditPositioningStrategy.settings.container = this.tbody.nativeElement;
+        this.rowEditPositioningStrategy.settings.target = row.element.nativeElement;
+        this.showRowEditingOverlay();
+        this.calculateRowChangesCount(row.rowID);
+    }
+
+    /**
+     * @hidden
+     */
+    public calculateRowChangesCount(rowID) {
+        if (!this.rowEditable) {
+            return;
+        }
+        const rowChanges = this.transactions.getAggregatedValue(rowID, false);
+        const rowChangedCount = rowChanges ? Object.keys(rowChanges).length : 0;
+        this.rowEditMessageValue = this.rowEditMessage.replace('{0}', rowChangedCount.toString());
+    }
+
+    /**
+     * @hidden
+     */
+    private endRowTransaction(commit?: boolean, closeOverlay?: boolean, row?: any) {
+        const rowInEdit = row ? row : this.gridAPI.get_row_inEditMode(this.id);
+        if (!rowInEdit) {
+            return;
+        }
+        const change = this.transactions.getAggregatedValue(rowInEdit.rowID, false);
+        const value = this.transactions.getAggregatedValue(rowInEdit.rowID, true);
+        const rowObj = rowInEdit ? this.getRowByIndex(rowInEdit.rowIndex) : null;
+        const emitter = commit ? this.onRowEditDone : this.onRowEditCancel;
+        emitter.emit({
+            newValue: Object.assign({}, this.data[rowObj.dataRowIndex], change),
+            oldValue: this.data[rowObj.dataRowIndex],
+            row: rowObj
+        });
+        this.transactions.endPending(commit);
+        if (commit && value && !isObjectEmpty(value) && !this.transactions.transactionsEnabled()) {
```

Comment:
With `enabled` change this should just check `this.transactions.enabled` after calling `endPending()`.